vltt/assignment1/Q1.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+#include <algorithm>
+#include <vector>
+#include <iomanip>
+#include <string>
+#include <unordered_set>
+#include <unordered_map>
+
+//is_case_sensitive == 1 -  sensitive, is_case_sensitive == 0 - insensitive
+
+
+void split_by_word(std::string &sentence, std::unordered_map<std::string, int> &word_counts, bool is_case_sensitive) {
```

Comment:
You should always pass non-trivial (anything other than a single numeric value or a boolean) objects as `const &`. The benefit of `const` is that it does not allow you to modify the passed-in object, creating a nice contract between the caller and the function that whatever is passed in will be left untouched. 

It's actually expected that whatever is not marked as `const` in the function signature WILL be modified and you would typically either copy the original value before calling such a function or only use throw-away values you don't care about.

So this line should be:

`void split_by_word(const std::string &sentence, std::unordered_map<std::string, int> &word_counts, bool is_case_sensitive) {`